doc/coq.rst

```diff
@@ -0,0 +1,392 @@
+  everything to compile correctly (this corresponds to the ``-Q`` flag for Coq).
+  
+  As of today, we only support composition with libraries defined in the same
+  scope (i.e. under the same :ref:`dune-project` domain). This restriction will
+  be lifted in the future. Note that composition with the Coq standard library
+  is supported, but in this case the ``Coq`` prefix has been made available in a
+  qualified way, since :ref:`Coq lang 0.2<coq-lang>`.
+
+- You can enable the production of Coq's native compiler object files by setting
+  ``<coq_native_mode>`` to ``native``. This passes ``-native-compiler on`` to
+  Coq and install the corresponding object files under ``.coq-native``, when in
+  the ``release`` profile. The regular ``dev`` profile skips native compilation
+  to make the build faster. This has been available since :ref:`Coq lang
+  0.3<coq-lang>`.
+
+  Please note: support for ``native_compute`` is **experimental** and requires a
+  version of Coq later than 8.12.1. Furthermore, dependent libraries *must* be
+  built with the ``(mode native)`` enabled. In Addition to that, Coq must be
+  configured to support native compilation. Dune explicitly disables the
+  generation of native compilation objects when ``(mode vo)`` is enabled,
+  irrespective of the configuration of Coq. This will be improved in the future.
+
+.. _include-subdirs-coq:
+
+Recursive Qualification of Modules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you add:
+
+.. code:: scheme
+
+    (include_subdirs qualified)
+
+to a :ref:`dune file<dune-files>`, Dune considers all the modules in the
+directory and its subdirectories, adding a prefix to the module name in the
+usual Coq style for subdirectories. For example, file ``A/b/C.v`` becomes the
+module ``A.b.C``.
+
+Limitations
+~~~~~~~~~~~
+
+- ``.v`` files always depend on the native OCaml version of the Coq binary and
+  its plugins, unless the natively compiled versions are missing.
+
+.. _limitation-mlpack:
+
+- A ``foo.mlpack`` file must the present in directories of locally defined
+  plugins for things to work. This is a limitation of ``coqdep``.
+
+- Building a theory and a plugin in the same directory can lead to issues with
+  the presence of the META file. We recommend the following:
+  - A separate directory for the files of each :ref:`coq-theory` stanza defined.
+  - A separate directory for source files of a plugin.
+
+.. _coq-lang:
+
+Coq Language Version
+~~~~~~~~~~~~~~~~~~~~
+
+The Coq lang can be modified by adding the following to a :ref:`dune-project`
+file:
+
+.. code:: scheme
+
+    (using coq 0.3)
+
+The supported Coq language versions (not the version of Coq) are:
+
+- ``0.1``: Basic Coq theory support.
+- ``0.2``: Support for the ``theories`` field and composition of theories in the
+  same scope,
+- ``0.3``: Support for ``(mode native)`` requires Coq >= 8.10 (and Dune >= 2.9
+  for Coq >= 8.14).
+
+.. _coq-lang-1.0:
+
+Coq Language Version 1.0
+~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
```suggestion
Coq Language Version 1.0 (planned)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Is 1.0 planned or experimental/under development?  i.e., will dune accept "1.0" as the language version _today_?
The challenge in describing functionality that's not yet complete are a) it may still change, so you need an extra pass on the doc and b) users may misread the doc and expect the future items already work.